solver/src/solver/http_solver.rs

```diff
@@ -75,6 +76,7 @@ pub struct HttpSolver {
 impl HttpSolver {
     #[allow(clippy::too_many_arguments)]
     pub fn new(
+        name: &'static str,
         account: Account,
         base: Url,
         api_key: Option<String>,
```

Comment:
General rust question:

How do we decide between when to use `String` or, in this case, `&'static str`?

Reply:
String is heap-allocated and potentially mutable. None is needed here.